modules/price/src/Plugin/Field/FieldFormatter/PriceCalculatedFormatter.php

```diff
@@ -84,11 +84,7 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function viewElements(FieldItemListInterface $items, $langcode) {
-    $currency_codes = [];
-    foreach ($items as $delta => $item) {
-      $currency_codes[] = $item->currency_code;
-    }
-    $currencies = $this->currencyStorage->loadMultiple($currency_codes);
+    $currencies = $this->currencyStorage->loadMultiple();
```

Comment:
I suggest to remove foreach at all. The problem is that in $item objects there is the currency code that is set for the commerce_price field. But if in the resolver I want to show the price in another currency? I tested this for the shop with multicurrency. I don't have price conversion and I created another price field in different currency. My resolver takes the price from the newly created field, but the price has other currency code. 

So when you collect currency codes in the loop as it is now, you collect the currency code that is set for the default field. And if the field is not multivalued (and it is not) $items will consists only of 1 item with 1 currency_code. So there is no way that formatter would show the price from the resolver in any other currency than in the currency that is set in the default price field.